frappe/client.py

```diff
@@ -405,13 +405,16 @@ def validate_link(doctype: str, docname: str, fields=None):
 	if not isinstance(docname, str):
 		frappe.throw(_("Document Name must be a string"))
 
-	if doctype != "DocType" and not (
-		frappe.has_permission(doctype, "select") or frappe.has_permission(doctype, "read")
-	):
-		frappe.throw(
-			_("You do not have Read or Select Permissions for {}").format(frappe.bold(doctype)),
-			frappe.PermissionError,
-		)
+	if doctype != "DocType":
+		parent_doctype = frappe.db.get_value(doctype, docname, "parenttype")  # needed for links to child rows
```

Comment:
One extra query for 0.01% use cases where it's required. Do this only if direct permission checks fail?

Reply:
Or possibly something even simpler, not sure why this "feels" unnecessary but there's something weird about it.

Reply:
Also, this will just fail if it's not a child doctype. 

---

@khushi8112 can you think of some better alternative... why do we need client-side selection of child table link that also gets validated?

Reply:
> Also, this will just fail if it's not a child doctype.

Huh, weird... I did test this on purpose and didn't see any failure.

Reply:
@barredterra maybe you have a site that predates this: https://github.com/frappe/frappe/pull/15411